### DISCUSSION TITLE
Using RandomLinkSplit + NeighborLoader.

### DISCUSSION THREAD

**Discussion by Yoloniverse** [pyg-team/pytorch_geometric#4024] [Q&A] [answered]
To do a link prediction task for heterodata, I want to use both ```RandomLinkSplit``` and ```NeighborLoader```. My dataset does not have masking. Therefore it is easier to use ```RandomLinkSplit```. But when I want to use ```NeighborLoader```, I should provide ```input_nodes``` in a form of input_nodes=('node_type', hetero_data['node_type'].train_mask). How can I provide train_mask here when ```RandomLinkSplit``` does not return ```train_mask```? Maybe it would be better to provide the mask when doing ```RandomLinkSplit```.

And for now as far as I understand, the ```NeighborLoader``` is based on node level tasks. Would you add update about ```NeighborLoader``` for ```link prediction```? Thank you.

**Comment by rusty1s** [comment, marked as answer]
Please see and comment on my newly created issue in https://github.com/pyg-team/pytorch_geometric/issues/4026, which tries to integrate link-level prediction support to `NeighborLoader`. I will try to integrate this feature ASAP. Currently, this can only be achieved by making use of some workarounds  (also linked in the issue).

Also note that it is well possible to pass either masks or indices to `input_nodes`.

**Reply by Yoloniverse** [reply]
I really appreciate you to create the issue #4026! Thank you.
